compiler-core/src/analyse.rs

```diff
@@ -1285,17 +1286,13 @@ impl<'a, A> ModuleAnalyzer<'a, A> {
         let type_ = fn_(arg_types, return_type);
         let _ = self.hydrators.insert(name.clone(), hydrator);
 
-        let external = target_function_implementation(
-            environment.target,
-            external_erlang,
-            external_javascript,
-        );
-        let (impl_module, impl_function) = implementation_names(external, &self.module_name, name);
         let variant = ValueConstructorVariant::ModuleFn {
             documentation: documentation.as_ref().map(|(_, doc)| doc.clone()),
-            name: impl_function,
+            name: name.clone(),
             field_map,
-            module: impl_module,
+            external_erlang: external_erlang.clone().map(|(m, f, _)| (m, f)),
+            external_javascript: external_javascript.clone().map(|(m, f, _)| (m, f)),
```

Comment:
Map then clone. Cloning then mapping results in more being cloned than is needed to be.